src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3202,14 +3206,16 @@ internal virtual ImmutableArray<LabelSymbol> Labels
                 methodName,
                 arity: 0,
                 basesBeingResolved: null,
-                options: LookupOptions.Default,
+                options: lookupOptions,
                 originalBinder: this,
                 diagnose: false,
                 useSiteDiagnostics: ref useSiteDiagnostics);
 
             diagnostics.Add(syntaxExpr, useSiteDiagnostics);
 
-            if (!lookupResult.IsMultiViable)
+            MethodSymbol patternMethod = null;
+
+            if (!lookupResult.IsMultiViable && !searchExtensionMethods)
             {
                 ReportPatternMemberLookupDiagnostics(lookupResult, patternType, methodName, syntaxExpr, warningsOnly, diagnostics, messageID);
```

Comment:
The `return null` on line 3231 looks suspicious -- I don't follow why the existence of a property with the Dispose name should disallow calling an identical Dispose method. Maybe this is a question we need to take to LDM